pkg/plugin/v3/webhook.go

```diff
@@ -106,9 +113,21 @@ func (p *createWebhookPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
 		return nil, fmt.Errorf("unable to load boilerplate: %v", err)
 	}
 
+	// Load the requested plugins
+	plugins := make([]model.Plugin, 0)
+	switch strings.ToLower(p.pattern) {
+	case "":
+		// Default pattern
+	case AddonPlugin:
+		plugins = append(plugins, &addon.Plugin{})
+	default:
+		return nil, fmt.Errorf("unknown pattern %q", p.pattern)
+	}
+
 	// Create the actual resource from the resource options
 	res := p.resource.NewResource(p.config, false)
-	return scaffolds.NewWebhookScaffolder(p.config, string(bp), res, p.defaulting, p.validation, p.conversion), nil
+	return scaffolds.NewWebhookScaffolder(p.config, string(bp), res, p.defaulting, p.validation, p.conversion, plugins),
+		nil
```

Comment:
It did not change the testdata project-v3-addon. Do we really add the that for the addon?

Reply:
Addon is not generating the webhooks in the testdata.

Reply:
WDYT about we add some mock for that so?

Reply:
Yes, we can add the webhook for the addons as well.

Reply:
I did not see a testdata update for add-on. Did you make that already?

Reply:
There is no webhook creation supported for addon right now. I need to make the kb create webhook command to accept --pattern addon.

Reply:
Catcher, we need to make the addon --partner to be a plugin as well. And then, after that, we could work to make the addon scaffold also support webhook if possible. So, it is fine 👍 Really thank you for check that.